docs/tutorials/05_DataAssociation-Clutter.py

```diff
@@ -0,0 +1,226 @@
+        timestamp=start_time+timedelta(seconds=k)))
+
+# %%
+# And create the Kalman predictor.
+
+from stonesoup.predictor.kalman import KalmanPredictor
+predictor = KalmanPredictor(transition_model)
+
+# %%
+# Simulating clutter
+# ------------------
+# Next, generate some measurements with uniform probability of detection, and add in some clutter
+# at each time-step. We use the :class:`~.TrueDetection` and :class:`~.Clutter` subclasses of
+# :class:`~.Detection` to help with discerning data types later in plotting. We'll introduce the
+# possibility that, at any time-step, our sensor receives no true detection from the target.
+
+import numpy as np
+
+from scipy.stats import uniform
+
+from stonesoup.types.detection import TrueDetection
+from stonesoup.types.detection import Clutter
+from stonesoup.models.measurement.linear import LinearGaussian
+measurement_model = LinearGaussian(
+    ndim_state=4,
+    mapping=(0, 2),
+    noise_covar=np.array([[0.75, 0],
+                          [0, 0.75]])
+    )
+all_measurements = []
+for state in truth:
+    measurement_set = set()
+
+    # Generate actual detection from the state with a 10% chance that no detection is received.
+    if np.random.rand() <= 0.9:
+        measurement = measurement_model.function(state, noise=True)
+        measurement_set.add(TrueDetection(state_vector=measurement,
+                                          groundtruth_path=truth,
+                                          timestamp=state.timestamp))
+
+    # Generate clutter at this time-step
+    truth_x = state.state_vector[0]
+    truth_y = state.state_vector[2]
+    for _ in range(np.random.randint(10)):
+        x = uniform.rvs(truth_x - 10, 20)
+        y = uniform.rvs(truth_y - 10, 20)
+        measurement_set.add(Clutter(np.array([[x], [y]]), timestamp=state.timestamp))
+
+    all_measurements.append(measurement_set)
+
+# %%
+# And create the Kalman updater.
+
+from stonesoup.updater.kalman import KalmanUpdater
+updater = KalmanUpdater(measurement_model)
+
+# %%
+# Plot the ground truth and measurements with clutter.
+
+from matplotlib import pyplot as plt
+fig = plt.figure(figsize=(10, 6))
+ax = fig.add_subplot(1, 1, 1)
+ax.set_xlabel("$x$")
+ax.set_ylabel("$y$")
+ax.set_ylim(0, 25)
+ax.set_xlim(0, 25)
+
+ax.plot([state.state_vector[0] for state in truth],
+        [state.state_vector[2] for state in truth],
+        linestyle="--",)
+
+for set_ in all_measurements:
+    # Plot actual detections.
+    ax.scatter([state.state_vector[0] for state in set_ if isinstance(state, TrueDetection)],
+               [state.state_vector[1] for state in set_ if isinstance(state, TrueDetection)],
+               color='b')
+    # Plot clutter.
+    ax.scatter([state.state_vector[0] for state in set_ if isinstance(state, Clutter)],
+               [state.state_vector[1] for state in set_ if isinstance(state, Clutter)],
+               color='y',
+               marker='2')
+fig
```

Comment:
Creating figure and having fig in same block causes plot to be shown twice in notebooks.